.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -24,6 +24,16 @@ Enter your extended release note in the block below.
 
 #### Additional Notes for your reviewer:
 
+##### Review Checklist:
```

Comment:
Can we add a bullet point for updating the kapp-controller dev.md doc if a change is development related?